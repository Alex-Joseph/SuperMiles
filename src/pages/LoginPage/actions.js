import { actionTypes } from './constants';


export const authenticateUser = (username, password, history) => (dispatch) => {
  if (username === 'username' && password === 'password') {
    dispatch(loginSuccess())
    history.push('/');
  } else {
    dispatch(loginFailure())
  }
};

export const logoutUser = history => (dispatch) => {
  history.push('/login');
  dispatch(logout())
};

export const loginSuccess = () => ({
  type: actionTypes.LOGIN_SUCCESS,
});

export const loginFailure = () => ({
  type: actionTypes.LOGIN_FAILURE,
});

export const logout = () => ({
  type: actionTypes.LOGOUT,
});
