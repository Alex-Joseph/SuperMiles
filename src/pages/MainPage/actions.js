import { actionTypes } from './constants';


export const postTransaction = (balance, product, history) => (dispatch) => {
  if (balance >= product.cost) {
    dispatch(updateBalance(balance - product.cost, product.desc));
    history.push('/receipt');
  } else {
    dispatch(insufficientBalance());
  }
};

export const updateBalance = (newBalance, desc) => ({
  type: actionTypes.UPDATE_BALANCE,
  balance: newBalance,
  transaction: { desc },
});

export const insufficientBalance = () => ({
  type: actionTypes.INSUFFICIENT_BALANCE,
});
