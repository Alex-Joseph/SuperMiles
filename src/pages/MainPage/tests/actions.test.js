import { postTransaction, insufficientBalance, updateBalance } from '../actions';


describe('action creators', () => {
  it('creates a postTransaction action', () => {
    expect(postTransaction()).toMatchSnapshot();
  });

  it('creates a updateBalance action', () => {
    expect(updateBalance()).toMatchSnapshot();
  });

  it('creates a insufficientBalance action', () => {
    expect(insufficientBalance()).toMatchSnapshot();
  });
});
