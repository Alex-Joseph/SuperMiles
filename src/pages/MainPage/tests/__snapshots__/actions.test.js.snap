// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators creates a insufficientBalance action 1`] = `
Object {
  "type": "INSUFFICIENT_BALANCE",
}
`;

exports[`action creators creates a postTransaction action 1`] = `[Function]`;

exports[`action creators creates a updateBalance action 1`] = `
Object {
  "balance": undefined,
  "transaction": Object {
    "desc": undefined,
  },
  "type": "UPDATE_BALANCE",
}
`;
